src/Symfony/Component/DomCrawler/AbstractUriElement.php

```diff
@@ -105,6 +127,24 @@ protected function canonicalizePath(string $path): string
      * @param \DOMElement $node A \DOMElement instance
      *
      * @throws \LogicException If given node is not an anchor
+     *
+     * @deprecated since Symfony 7.1, use `setDomNode()` instead
      */
     abstract protected function setNode(\DOMElement $node): void;
+
+    /**
+     * Sets current \DOMElement or \DOM\Element instance.
+     *
+     * @param \DOMElement|\DOM\Element $node A \DOMElement or \DOM\Element instance
+     *
+     * @throws \LogicException If given node is not an anchor
+     */
+    protected function setDomNode(\DOMElement|\DOM\Element $node): void
+    {
+        $this->domNode = $node;
+
+        if ($node instanceof \DOMElement) {
+            $this->setNode($node);
```

Comment:
The way this abstract class is implemented means that all child classes are broken when using the new extension if they don't override `setDomNode` instead of `setNode`. So this method should trigger a deprecation warning asking to override it (and in 8.0, it will become abstract)